app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,11 @@
+class CompaniesController < ApplicationController
+
+  def index
+    @companies = Company.all
+  end
+
+  def show
+    @company = Company.find(params[:id])
+    @items = Item.items_by_company(@company[:shortname])
```

Comment:
your schema shows that an item has a company_id. can you build the relationship on the model for a company has_many items and an item belongs_to a company, then just pass @items as @company.items?